doc/coq.rst

```diff
@@ -0,0 +1,62 @@
+.. _coq-main:
+
+******
+Coq
+******
+
+Dune is also able to build Coq developments. A Coq project is a mix of
+Coq ``.v`` files and (optionally) OCaml libraries (in which case we say the project is a *Coq plugin*). To enable
+Coq support in a dune project, the language version should be selected
+in the ``dune-project`` file. For example:
+
+.. code:: scheme
+
+    (using coq 0.1)
+
+This will enable support for the ``coqlib`` stanza in the current project. If the
+language version is absent, dune will automatically add this line with the
+latest Coq version to the project file once a ``(coqlib ...)`` stanza is used anywhere.
+
+
+Basic Usage
+===========
+
+The basic form for defining Coq libraries is very similar to the OCaml form:
+
+.. code:: scheme
+
+    (coqlib
+     (name <module_prefix>)
+     (synopsis <text>)
+     (modules <ordered_set_lang>)
+     (flags <coq_flags>))
+
+The stanza will build all `.v` files on the given directory.
+The semantics of fields is:
+- ``<module_prefix>>`` will be used as the default Coq library prefix
+  ``-R``
```

Comment:
I am not sure I understand the choice of `-R` over `-Q` which was IIMN the recommended choice these days.

Reply:
I think virtually no library will compile with -Q, note that once installed -Q is implicit as of today given that coqtop includes the top level user-contribs.

We need to discuss the library model indeed, but I'll open an issue for that as there are many complications. For now this PR and 1555 will just reach the point single libraries can be handled properly as in coq makefile.

Reply:
> I suggest you leave this for now and just create an issue. We'll address this in a subsequent PR.

Understood, thanks!

Reply:
Many libraries actually rely on `-Q` as of today and it would be good that there is no regression for these libraries (as coq_makefile supports the two options). That being said, I can understand that this is still experimental support and could be seen as a later refinement.